src/Components/Server/src/Builder/ComponentEndpointRouteBuilderExtensions.cs

```diff
@@ -12,6 +12,22 @@ namespace Microsoft.AspNetCore.Builder
     /// </summary>
     public static class ComponentEndpointRouteBuilderExtensions
     {
+        /// <summary>
+        /// Maps the SignalR <see cref="ComponentHub"/> to the path <paramref name="path"/> and associates
+        /// the component <typeparamref name="TComponent"/> to this hub instance as the given DOM <paramref name="selector"/>.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/>.</param>
+        /// <returns>The <see cref="IEndpointConventionBuilder"/>.</returns>
+        public static IEndpointConventionBuilder MapComponentHub(this IEndpointRouteBuilder endpoints)
```

Comment:
Does signalR return a more specific type than `IEndpointConventionBuilder`? We should return that here if possible.

Reply:
Nope. It returns IEndpointConventionBuilder. I agree, if they return something more specific we should do too